src/XrdTpc/XrdTpcTPC.cc

```diff
@@ -132,7 +136,11 @@ int TPCHandler::opensocket_callback(void *clientp,
   {XrdNetAddr thePeer(&(aInfo->addr));
     rec->isIPv6 =  (thePeer.isIPType(XrdNetAddrInfo::IPv6)
                     && !thePeer.isMapped());
-    // Register the socket to the packet marking manager
+    // Connect the socket and register it to the packet marking manager
+    if(connect(fd, (struct sockaddr*)&(aInfo->addr), aInfo->addrlen) ==
+       -1) {
```

Comment:
The `-1) {` could go on the same line above, not worth breaking up the line for this, but this is fairly minor, I'm fine if it stays like this.

Reply:
Note this turns the callback from non-blocking to blocking without a timeout.

Blocking behavior is probably not so bad -- a separate event loop is run for each transfer so it's only blocking itself -- but the lack of timeout is problematic.

My $0.02: instead of making this work completely non-blocking (I've done this before in a different project: you basically have to fail the operation, create the connection in the event loop outside curl, then retry and pull from the curl cache.  It's ugly for a use case where not needed), simply add a timeout to the connection here.